framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -2327,4 +2232,37 @@ public AnnotatedTypeMirror getDummyAssignedTo(ExpressionTree expressionTree) {
         }
         return null;
     }
+
+    /**
+     * Unboxes a type if needed all the way to a primitive integral.
+     *
+     * @param type TypeMirror to unbox
+     * @return the TypeKind once unboxed
+     */
+    private static TypeKind resolveToPrimitiveIntegralTypeKind(TypeMirror type) {
+        TypeKind typeKind = NumberUtils.unboxPrimitive(type);
+        if (isPrimitiveIntegral(typeKind)) {
+            return typeKind;
+        }
+        throw new BugInCF(type.toString() + " expected to be an integral type.");
```

Comment:
This error message is confusing.  If bad client code passes in (say) String, the error is not that String is expected to be an integral type.  The error is that the argument should have been an integral type.  Could you reword it so it doesn't take reading a couple of times to understand?

Reply:
See https://github.com/typetools/checker-framework/pull/2942#issuecomment-560397137